### PR TITLE
Group Color Fixes

### LINKED ISSUE #429: Transaction Using Group Color Doesn't Update On Group Color Change
From Matrix Room:
> If I create a transaktion and the transaction is every month (and I do this in the settings of the transaction) and then use the group color. After I create the transaction, I edit the group color. And then, the color isn't automatically updated and I have to edit the transaction, do nothing, save. And then the color is updated.

### PR DESCRIPTION
**Pull request by nlogozzo**
This PR fixes an issue where transactions using a group's color would not update when said group's color was changed. This PR also includes a change that updates the `Ungrouped` row's color when it is changed in the configuration.

Closes #429

### CODE REVIEW

**Comment by DaPigGuy** [comment]
Transaction color should also update if the group gets deleted

**Comment by nlogozzo** [comment]
> Transaction color should also update if the group gets deleted

Done :)

**Review by fsobolev — CHANGES_REQUESTED** [review]
Something goes wrong, can't find where exactly... On group color change the color of belonging transaction gets updated, but also its currency gets changed to unknown

Снимок экрана от 2023-04-09 19-20-47

File: NickvisionMoney.Shared/Models/Account.cs [hunk 1]
```diff
@@ -815,13 +817,19 @@ public async Task<bool> DeleteGroupAsync(uint id)
                 if (pair.Value.GroupId == id)
                 {
                     pair.Value.GroupId = -1;
+                    if (pair.Value.UseGroupColor)
+                    {
+                        pair.Value.UseGroupColor = false;
+                        pair.Value.RGBA = defaultTransactionColor;
```
Comment (fsobolev) [inline NickvisionMoney.Shared/Models/Account.cs:823]: do we really need to change RGBA to default here? At least in GNOME RGBA is always saved from ColorDialogButton, even when UseGroupColor is true. It also can be other than the default value if a user changed it, so I think it does make sense to use it without changing to default.

**Review by fsobolev — CHANGES_REQUESTED** [review]

**Comment by nlogozzo** [comment]
> Something goes wrong, can't find where exactly... On group color change the color of belonging transaction gets updated, but also its currency gets changed to unknown

Fixed in the latest commit. Somehow `CultureForNumberString` had a `.Parent` attached to it making it use the parent culture instead of the actual culture itself 😅 VS IntelliSense probably suck it in on me haha

File: NickvisionMoney.Shared/Models/Account.cs (hunk 1 above)
Comment (nlogozzo) [reply NickvisionMoney.Shared/Models/Account.cs:823]: Yeah you are right, i'll remove that line.

**Review by fsobolev — APPROVED** [review]

**Pull request merged**